pkg/credentials/iam_aws.go

```diff
@@ -116,11 +117,20 @@ func (m *IAM) Retrieve() (Value, error) {
 			roleSessionName: os.Getenv("AWS_ROLE_SESSION_NAME"),
 		}
 
-		stsWebIdentityCreds, err := creds.Retrieve()
-		if err == nil {
-			m.SetExpiration(creds.Expiration(), DefaultExpiryWindow)
+		// FIXME: a more comprehensive fix is needed for all
+		// providers with an expontential retry and logging.
+		//
+		// Loop until getting a valid credentials
+		for {
+			stsWebIdentityCreds, err := creds.Retrieve()
+			if err == nil {
+				m.SetExpiration(creds.Expiration(), DefaultExpiryWindow)
+				return stsWebIdentityCreds, nil
+			}
+
+			log.Printf("unable to fetch STS credentials: %s", err.Error())
```

Comment:
We can't log like this @vadmeste .. it needs to be a custom logger.

Reply:
> it needs to be a custom logger.

This is more complex than it looks. That is why I wanted a separate PR for this, since we are also going to release today, I did not want to risk it.